ml4h/explorations.py

```diff
@@ -691,6 +699,206 @@ def infer_with_pixels(args):
+                logging.info(true_outliers['sample_id'].to_list())
+                logging.info(f'sample_ids where {col} is zero in the model segmentation:')
+                logging.info(pred_outliers['sample_id'].to_list())
+            elif i == 'filter_outliers':
+                plot_data = plot_data[plot_data.true != 0]
+                plot_data = plot_data[plot_data.pred != 0]
+            plot_data = plot_data.drop('sample_id', axis=1)
+
+            plt.figure()
+            g = lmplot(x='true', y='pred', data=plot_data)
+            ax = plt.gca()
+            title = col.replace('_', ' ')
+            ax.set_xlabel(f'{title} T1 Time (ms) - Manual Segmentation')
+            ax.set_ylabel(f'{title} T1 Time (ms) - Model Segmentation')
+            if i == 'all':
+                min_value = -50
+                max_value = 1300
+            elif i == 'filter_outliers':
+                min_value, max_value = plot_data.min(), plot_data.max()
+                min_value = min([min_value['true'], min_value['pred']]) - 100
+                max_value = min([max_value['true'], max_value['pred']]) + 100
+            ax.set_xlim([min_value, max_value])
+            ax.set_ylim([min_value, max_value])
+            res = stats.pearsonr(plot_data['true'], plot_data['pred'])
+            conf = res.confidence_interval(confidence_level=0.95)
+            text = f'Pearson Correlation Coefficient r={res.statistic:.2f},\n95% CI {conf.low:.2f} - {conf.high:.2f}'
+            ax.text(0.25, 0.1, text, transform=ax.transAxes)
+            if i == 'all':
+                postfix = ''
+            elif i == 'filter_outliers':
+                postfix = '_no_outliers'
+            logging.info(f'{col} pearson{postfix} {res.statistic}')
+            figure_path = os.path.join(
+                output_folder, id, f'{col}_{id}_{input_name}_{output_name}{postfix}.png',
+            )
+            plt.savefig(figure_path)
+
+def infer_stats_from_segmented_regions(args):
+    """
+    Computes .tsv files of intensity means, medians and standard deviations within predicted segmentations for
+    a given list of structures of interest. If ground truth segmentations are available, computes the same
+    statistics within them, as well as scatter plots that compare median intensities within predicted and
+    ground truth segmentations.
+    """
+    assert(args.batch_size == 1, 'no support here for iterating over larger batches')
+    assert(len(args.tensor_maps_in) == 1, 'no support here for multiple input maps')
+    assert(len(args.tensor_maps_out) == 1, 'no support here for multiple output channels')
+
+    tm_in = args.tensor_maps_in[0]
+    tm_out = args.tensor_maps_out[0]
+    assert(tm_in.shape[-1] == 1, 'no support here for stats on multiple input channels')
+
+    # don't filter datasets for ground truth segmentations if we want to run inference on everything
+    # TODO HELP - this isn't giving me all 56K anymore
```

Comment:
I thought you figured this issue out? if so remove comment.

Reply:
This was removed on the other branch (dp_kidney)